Tests/StatsdClientTests/Stub|Mock|Fake|Spy/MockStorage.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  MockStorage.swift
+//  StatsdClient
+//
+//  Created by Nghia Tran on 10/15/17.
+//  Copyright © 2017 StatsdClient. All rights reserved.
+//
+
+import Foundation
+@testable import StatsdClient
+
+class MockStorage<ItemType>: MemoryStorage<ItemType> {
```

Comment:
What is the purpose of this class?

Reply:
StatsD's initialization is hard-coding concreted `MemoryStorage` class instead Generic `Storage` protocol.
Furthermore, MemoryStorage class is an external resource, and it could contain unexpected flaws. 
```swift
init(transport: Transport, storage: MemoryStorage<Metric>, flush: Flushable)
```
It's impossible to inject `MockStorage`, which is adopted `Storage` protocol for testing.

As a result, I must create MockStorage, which subclass MemoryStorage class and override all methods and implement the simple code in order to reduce the risk of errors.

Then I could test StatsD easily.

Reply:
The problem is `MockStorage` and `MemoryStorage` don't have much differences, beside the threading thing. 
It's fine to use `MemoryStorage` to test for now, later on we'll add Mock if needed.